tests/components/azure_data_explorer/test_init.py

```diff
@@ -265,6 +267,23 @@ async def test_event(
     mock_managed_streaming.add.assert_not_called()
 
 
+async def test_json_dumps_raises_TypeError(
+    hass: HomeAssistant, entry_managed: MockConfigEntry, mock_managed_streaming: Mock
+) -> None:
+    """Test listening to events from Hass. and getting an event with a newline in the state."""
+
+    with mock.patch.object(json, "dumps", side_effect=TypeError):
```

Comment:
I don't like to patch json, is there a more real use case that can trigger this?

Reply:
The usecase that I have, that actually is triggering, is having a state with a picture element. when using this, the I get an error that its too long a state, despite that is what I see in the real world. this is why I mocked the Error

Reply:
What do you mean with picture element? an `image.*` entity?

Reply:
This is the event that is failing, because its not serializable:

entity_id='sensor.affalddk_adress_farligt_affald' state='69' attributes={'date': '2024-08-29', 'date_long': 'Torsdag d. 29-08-2024', 'date_short': 'Tor d. 29/08', 'description': 'Rød boks (1 stk.)', 'duration': 'Om 69 dage', 'entity_picture': 'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSI2ODAuMzIiIGhlaWdodD0iNjgwLjMyIj48cGF0aCBkPSJNNTEwMi4zNiAwSDB2NTEwMi4zNmg1MTAyLjM2eiIgc3R5bGU9ImZpbGw6I2U5MWQyYztmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIgdHJhbnNmb3JtPSJtYXRyaXgoLjEzMzMzIDAgMCAtLjEzMzMzIDAgNjgwLjMyKSIvPjxwYXRoIGQ9Im0yNjAyLjIzIDIzNDUuMTYtMTAxLjg3LjE0LTUxLjIgODE2LjI0aDIwNC4xem0tNTEuMDItMzA0LjQ0Yy01Ni4zNiAwLTEwMi4wNSA0NS42OS0xMDIuMDUgMTAyLjA1czQ1LjY5IDEwMi4wNSAxMDIuMDUgMTAyLjA1IDEwMi4wNS00NS42OSAxMDIuMDUtMTAyLjA1LTQ1LjY5LTEwMi4wNS0xMDIuMDUtMTAyLjA1bTEyMDAuNjgtMTQ3LjI0LjAxLjAyLTExNDAuNDkgMTg5OC45OWMtMTIuMzEgMjAuMzQtMzQuNjYgMzMuOTMtNjAuMiAzMy45My0yNy4wMyAwLTUwLjQ4LTE1LjIzLTYyLjI0LTM3LjU2TDEzNTAuMjEgMTg5M2wuMDEtLjAxYy03Ljg5LTEyLjQ3LTExLjk2LTI3LjU3LTEwLjU0LTQzLjcgMy4yMy0zNi43NCAzNi4xOC02My44MSA3My4wNy02My44MWgyMjc2LjgzYzM3LjA1IDAgNzAuMTQgMjcuMjggNzMuMTkgNjQuMiAxLjM1IDE2LjItMi44NiAzMS4zNC0xMC44OCA0My44IiBzdHlsZT0iZmlsbDojZmZmO2ZpbGwtb3BhY2l0eToxO2ZpbGwtcnVsZTpub256ZXJvO3N0cm9rZTpub25lIiB0cmFuc2Zvcm09Im1hdHJpeCguMTMzMzMgMCAwIC0uMTMzMzMgMCA2ODAuMzIpIi8+PC9zdmc+', 'last_update': '2024-06-21T10:18:27.519373+02:00', 'name': 'Farligt affald', 'unit_of_measurement': 'dage', 'attribution': 'Data delivered by Danish Municipalities', 'icon': 'mdi:recycle', 'friendly_name': 'Affalddk adress Farligt affald'} last_changed=datetime.datetime(2024, 6, 21, 0, 18, 27, 532508, tzinfo=TzInfo(UTC)) last_updated=datetime.datetime(2024, 6, 21, 8, 18, 27, 520653, tzinfo=TzInfo(UTC)) context=Context(id='01J0WY07J0ZSEWQ57XG327CMZ0')

Reply:
This is the direct output of another integration (affalddk from HACS), but as I see it, ADX needs to be able to automatically filter out stuff that for one reason or another is not serializable

Reply:
Oooh, but that means you just make a sensor that properly tests this.

Like I'd rather see tests where we properly test this than to rely on json.dumps to raise an issue. It will also help with regressions in the future

Reply:
I agree, but when I aplieed that exact string into the state change, I get an too long error, and the test fails without running. Im not sure for to test this without mocking the json.dumps?

Reply:
Create a fake string, that is not serializable?

Reply:
Tryed this: hass.states.async_set("sensor.test_sensor", b"non_serializable_event")

but this will not get json.dump to fail, but not within HA, it must be cast somewhere before reaching the json.dumps.

I might also be chasing chosts here, I have a hard time getting the exact error. the only problem I see is that with the above mentioned HACS integration, both this and Azure Eventhub Integration stalls, and stops sending more events. This is documented in this issue: https://github.com/home-assistant/core/issues/119812

But I really dont know how to deeply debug this issue. the only error I get is after the reset of the integration I get a "x number of messegas dropped" in the log, and I can see that the memory consumption of both ADX and AEH integrations skyrockets, as it accumulates more and more messeges in memory without sending them. And ofcurse no messages is recieved in Azure. When disabling or setting a filter for this affalddk integration, the problem disapears. 

SO the only thing I actually knows is that this integration is giving me the problems, and it looks very strange in its abnormal long state

Hope you can help me

Reply:
But there you're setting the state to a binary value what doesn't work. From your example I'm assuming that the binary info is the entity picture, so wouldn't setting the entity picture of an entity achieve the same?

Reply:
And maybe a good temporary solution could be ignoring certain fields when serializing